daemon/src/project.rs

```diff
@@ -0,0 +1,69 @@
+//! Project creation data and functions.
+
+use std::{io, path::Path};
+
+use librad::{
+    git::types::remote::Remote,
+    git_ext::{is_not_found_err, OneLevel, Qualified},
+    std_ext::result::ResultExt,
+};
+
+/// Module concerned with creating new projects and repositories.
+pub mod create;
+pub use create::{Create, Repo};
+
+/// Module concerned with checkout out working copies of projects, as git repositories.
+pub mod checkout;
+pub use checkout::Checkout;
+
+pub mod peer;
+pub use peer::Peer;
+
+/// Set the upstream of the given branch to the given remote.
+///
+/// This writes to the `config` directly. The entry will look like the following:
+///
+/// ```text
+/// [branch "main"]
+///     remote = rad
+///     merge = refs/heads/main
+/// ```
+fn set_upstream<Url>(
```

Comment:
Seems to be similar to other utilities better placed somewhere else.

Reply:
Should be addressed in https://github.com/radicle-dev/radicle-link/issues/587